src/rules/noAsyncWithoutAwaitRule.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license
+ * Copyright 2018 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../index";
+import { codeExamples } from "./code-examples/noAsyncWithoutAwait.examples";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "Functions marked async must contain an await or return statement.";
+
+    public static metadata: Lint.IRuleMetadata = {
+        codeExamples,
+        description: Rule.FAILURE_STRING,
+        hasFix: false,
+        optionExamples: [true],
+        options: null,
+        optionsDescription: "Not configurable.",
+        rationale: "Cleaner code, can possibly reduce transpiled output",
```

Comment:
Please explain this in more detail. Someone reading this rule's page on palantir.github.io would leave with more questions than they started with.